apps/molgenis-components/src/components/forms/InputDecimal.vue

```diff
@@ -14,40 +15,45 @@
       :aria-describedby="id + 'Help'"
       :placeholder="placeholder"
       @keypress="handleKeyValidity"
-      @input="$emit('input', $event.target.value)"
+      @input="emitIfValid"
     />
   </FormGroup>
 </template>
 
 <script>
 import BaseInput from "./BaseInput.vue";
 import FormGroup from "./FormGroup.vue";
-import constants from "../constants.js";
-
-const { CODE_0, CODE_9, CODE_BACKSPACE, CODE_DELETE } = constants;
+import { isNumericKey } from "./utils/InputUtils";
 
 export default {
   extends: BaseInput,
   components: {
     FormGroup,
   },
-  props: {
-    errorMessage: { type: String, default: null },
+  computed: {
+    validDecimal() {
```

Comment:
errorMessage is a better name, since that is what is returned and how it is used in the html